web/router.go

```diff
@@ -91,24 +94,78 @@ func guiEngine(app *services.ChainlinkApplication) *gin.Engine {
 		basicAuth,
 	)
 
-	box := packr.NewBox("../gui/dist/")
+	box := guiBox()
+	boxList := box.List()
+
 	engine.NoRoute(func(c *gin.Context) {
-		if filepath.Ext(c.Request.URL.Path) == "" {
-			index, err := box.Open("index.html")
-			if err != nil {
-				logger.Warn(err)
-			}
+		path := c.Request.URL.Path
 
-			buf := new(bytes.Buffer)
-			buf.ReadFrom(index)
-			c.Data(http.StatusOK, "text/html; charset=utf-8", buf.Bytes())
+		if filepath.Ext(path) == "" {
+			matchedPath := matchPath(boxList, path, "/index.html")
+			if matchedPath != "" {
+				writeData(c, "text/html", box, matchedPath)
+			}
+		} else if filepath.Ext(path) == ".json" {
+			matchedPath := matchPath(boxList, path, "")
+			if matchedPath != "" {
+				writeData(c, "appliction/json", box, matchedPath)
+			}
 		}
 	})
 	engine.StaticFS("/", box)
 
 	return engine
 }
 
+func guiBox() packr.Box {
+	if flag.Lookup("test.v") == nil {
```

Comment:
Any suggestions on how to exclude this from the binary of production builds?

Reply:
@se3000 I can get the build constraint to work but you need to pass tags which is different to how the current implementation works. It would mean that we need to run `go test` with the tags option e.g.

```
go test ./... --tags=test
```

What are your thoughts on this? To improve the workflow we could alias it or create a test script that we use instead of `go test`

Reply:
I thought it came for free. Reading https://github.com/golang/go/issues/14668, it doesn't seem likely that anything else like this will exist in the language.

I guess at this point, I'd prefer to leave it in by default and have an optional release flag that we use in our Makefile for actual leaving out the functionality.

Reply:
It seems like it was a proposal that never got traction. There's some extra discussion over [here](https://github.com/golang/go/issues/21360) which has an interesting [suggestion](https://github.com/golang/go/issues/21360#issuecomment-322550450).

It's pretty much what you suggested in terms of having an optional release flag. We could just always set it on the CI tests.